### PR TITLE
(SIMP-6820) Fix environment section

### PR DESCRIPTION
**Pull request by pillarsdotnet**
Hoping this works...

### CODE REVIEW

File: manifests/conf.pp [hunk 1]
```diff
@@ -27,12 +32,22 @@
...
+    if !defined(Ini_setting['remove environment from main']) {
+      ini_setting { 'remove environment from main':
+        ensure  => absent,
+        path    => "${confdir}/puppet.conf",
+        section => 'main',
+        setting => 'environment'
+      }
+    }
+  }
+
```
Comment (pillarsdotnet) [inline manifests/conf.pp]: There should be a way to force-set `environment` within `main` if the user wishes to do so.  Perhaps add a `$pupmod::master::allow_environment_in_main` param that defaults to `false` ?

I'll try that...

Comment (pillarsdotnet) [reply manifests/conf.pp]: Pushed a commit for review.

File: manifests/init.pp [hunk 2]
```diff
@@ -140,38 +144,39 @@
...
+  String[0]                              $classfile                    = '$vardir/classes.txt',
+  Stdlib::AbsolutePath                   $confdir                      = $::pupmod::params::puppet_config['confdir'],
+  Boolean                                $daemonize                    = false,
+  Enum['md5','sha256']                   $digest_algorithm             = 'sha256',
+  Boolean                                $enable_puppet_master         = false,
+  Stdlib::AbsolutePath                   $environmentpath              = $::pupmod::params::puppet_config['environmentpath'],
+  Boolean                                $listen                       = false,
+  Stdlib::AbsolutePath                   $logdir                       = $::pupmod::params::puppet_config['logdir'],
+  Simplib::Port                          $masterport                   = 8140,
+  Boolean                                $remove_environment_from_main = true,
```
Comment (trevor-vaughan) [inline manifests/init.pp]: @pillarsdotnet I see what you're getting at here and agree with the concept but I'm not super keen on adding a 'workaround' variable to the API.

What do you think about instead changing `$set_environment` to the following:
* `true` => Set the environment and clean the environment out of `main`
* `no_clean` => Set the environment and do not clean the environment out of `main`
* `false` => Don't do anything

**Comment by trevor-vaughan** [comment]
:hourglass_flowing_sand: https://gitlab.com/simp/pupmod-simp-pupmod/pipelines/69500614

**Comment by trevor-vaughan** [comment]
Rebased

**Pull request merged**